File: pglite/src/pglite/contrib/btree_gin.ts
import type {
  Extension,
  ExtensionSetupResult,
  PGliteInterface,
} from '../interface'

const setup = async (_pg: PGliteInterface, _emscriptenOpts: any) => {
  return {
    bundlePath: new URL('btree_gin.tar.gz', import.meta.url),
  } satisfies ExtensionSetupResult
}

export const btree_gin = {
  name: 'btree_gin',
  setup,
} satisfies Extension

File: lib/src/main/java/io/github/hidekatsu_izuno/pglite_jdbc/pglite/contrib/btree_gin.java
package io.github.hidekatsu_izuno.pglite_jdbc.pglite.contrib;

import io.github.hidekatsu_izuno.pglite_jdbc.pglite.interface_.Extension;
import io.github.hidekatsu_izuno.pglite_jdbc.pglite.interface_.ExtensionSetupResult;
import java.net.URL;
import java.util.concurrent.CompletableFuture;

public final class btree_gin {
    public static final Extension<Object> btree_gin = new Extension<>();

    static {
        btree_gin.name = "btree_gin";
        btree_gin.setup = (_pg, emscriptenOpts, clientOnly) -> {
            var result = new ExtensionSetupResult<Object>();

            result.bundlePath = resource("btree_gin.tar.gz");
            return CompletableFuture.completedFuture(result);
        };
    }

    private static URL resource(String name) {
        var url = btree_gin.class.getClassLoader().getResource(name);
        if (url == null) {
            throw new RuntimeException("Extension bundle not found on classpath: " + name);
        }
        return url;
    }

    private btree_gin() {
    }
}
